forge-ai/src/main/java/forge/ai/ability/DamageDealAi.java

```diff
@@ -95,8 +95,8 @@ protected boolean canPlayAI(Player ai, SpellAbility sa) {
         final String damage = sa.getParam("NumDmg");
         int dmg = AbilityUtils.calculateAmount(source, damage, sa);
 
-        if (damage.equals("X") || sourceName.equals("Crater's Claws")) {
-            if (sa.getSVar(damage).equals("Count$xPaid") || sourceName.equals("Crater's Claws")) {
+        if (damage.equals("X") || source.getSVar("X").equals("Count$xPaid") || sourceName.equals("Crater's Claws")) {
```

Comment:
I don't understand, Fireball should have already passed the old checks?

Reply:
It looks like the card (as well as other similar cards) were updated, so the "damage" parameter no longer contained what was expected (e.g. NumDmg was pointing to Y, not to X, and X still had Count$xPaid). I'm not sure when it was rewritten and what else may have been rewritten (and to what extent), but at least for these two cards, the checks are now different.

Reply:
I see, I was only looking at Fireball
for Shatterskull Smashing, it has always been that way